branchandbound.js

```diff
@@ -0,0 +1,143 @@
+var utils = require('./utils')
+
+var maxTries = 1000000
```

Comment:
I have left 1M here; in my experience it's very fast anyway, even in javascript on browser/in node